### PR TITLE
Expose enable/disable debugging

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Expose enable/disable debugging when CLI is required as a library.
Includes:
* Unify `debug-livesync-service` and `livesync-service` into one class. Enable said class to attach a debugger at runtime during LiveSync
* Remove redundant `debugLivesync` property from `$config` and its every usage
* Remove redundant `debugStop` code in `android-debug-service`
* Raise events in the event of inability to start/stop the app on iOS
* Use `debugService` for debugging instead of platform-specific services.
* Instantiate a new platform-specific debug service for each device
* Implement `debugStop` for Android - cancel the port forwarding

Merge after https://github.com/telerik/mobile-cli-lib/pull/1001

Ping @rosen-vladimirov @TsvetanMilanov

### CODE REVIEW

**Comment by dtopuzov** [comment]
run ci

File: lib/services/livesync/livesync-service.ts [hunk 1]
```diff
@@ -120,6 +135,150 @@ export class LiveSyncService extends EventEmitter implements ILiveSyncService {
...
+		const projectData = this.$projectDataService.getProjectData(settings.projectDir);
+		let debugData = this.$debugDataService.createDebugData(projectData, { device: settings.deviceIdentifier });
+
+		// Of the properties below only `buildForDevice` and `release` are currently used.
+		// Leaving the others with placeholder values so that they may not be forgotten in future implementations.
+		const buildConfig: IBuildConfig = {
+			buildForDevice: !settings.isEmulator,
+			release: false,
+			device: settings.deviceIdentifier,
+			provision: null,
```
Comment (Mitko-Kerezov) [inline lib/services/livesync/livesync-service.ts]: This `buildConfig` isn't actually used for building - only for getting the last output path

File: lib/services/livesync/livesync-service.ts [hunk 2]
```diff
@@ -96,21 +101,31 @@ export class LiveSyncService extends EventEmitter implements ILiveSyncService {
 		return currentDescriptors || [];
 	}
 
-	protected async refreshApplication(projectData: IProjectData, liveSyncResultInfo: ILiveSyncResultInfo): Promise<void> {
+	private async refreshApplication(projectData: IProjectData, liveSyncResultInfo: ILiveSyncResultInfo, debugOpts?: IDebugOptions, outputPath?: string): Promise<void> {
+		const liveSyncProcessInfo = this.liveSyncProcessesInfo[projectData.projectDir];
+		const deviceDescriptor = _.find(liveSyncProcessInfo.deviceDescriptors, dd => dd.identifier === liveSyncResultInfo.deviceAppData.device.deviceInfo.identifier);
```
Comment (rosen-vladimirov) [inline lib/services/livesync/livesync-service.ts]: you can use:
```TypeScript
const deviceDescriptors = this.getLiveSyncDeviceDescriptors(projectData.projectDir);
const deviceDescriptor = _.find(deviceDescriptors, dd => dd.identifier === liveSyncResultInfo.deviceAppData.device.deviceInfo.identifier);
```

File: lib/services/livesync/livesync-service.ts [hunk 3]
```diff
@@ -96,21 +101,31 @@ export class LiveSyncService extends EventEmitter implements ILiveSyncService {
...
 			this.$logger.warn(msg);
+			if (!settings || !settings.shouldSkipEmitLiveSyncNotification) {
 			this.emit(LiveSyncEvents.liveSyncNotification, {
 				projectDir: projectData.projectDir,
 				applicationIdentifier: projectData.projectId,
 				deviceIdentifier: liveSyncResultInfo.deviceAppData.device.deviceInfo.identifier,
 				notification: msg
 			});
 		}
+		}
```
Comment (rosen-vladimirov) [inline lib/services/livesync/livesync-service.ts]: incorrect tabs

File: lib/services/livesync/livesync-service.ts [hunk 4]
```diff
@@ -123,6 +138,161 @@ export class LiveSyncService extends EventEmitter implements ILiveSyncService {
...
+			this.$logger.info(`To start debugging, open the following URL in Chrome:${EOL}${information}${EOL}`.cyan);
+		}
+	}
+
+	public enableDebugging(deviceOpts: IEnableDebuggingDeviceOptions[], debuggingAdditionalOptions: IDebuggingAdditionalOptions): Promise<void>[] {
+		return _.map(deviceOpts, d => this.enableDebuggingCore(d, debuggingAdditionalOptions));
+	}
+
+	private getDeviceDescriptor(deviceIdentifier: string, projectDir: string) {
+		const liveSyncProcessInfo: ILiveSyncProcessInfo = this.liveSyncProcessesInfo[projectDir] || <ILiveSyncProcessInfo>{};
```
Comment (rosen-vladimirov) [inline lib/services/livesync/livesync-service.ts]: again you can use `getLiveSyncDeviceDescriptors`

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

**Pull request merged**